dependency_graph/generate.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env python
+#
+# generate.py
+#
+# Python script to generate a DOT graph showing the dependency graph of
+# the components within the Ethereum webthree-umbrella project.
+#
+# See http://github.com/ethereum/webthree-umbrella for more info on webthree
+# See http://ethereum.org for more info on Ethereum
+#
+# Contributed by Bob Summerwill (bob@summerwill.net)
+
+import os
+import re
+
+assignmentPattern = "(set|eth_name)\((EXECUTABLE|LIBRARY) (.*)\)"
+dependencyPattern = "eth_use\((.*)\s+(OPTIONAL|REQUIRED)\s+(.*)\)"
+
+
+# Returns a string which lists all the dependency edges for a given
+# library or application, based on the declared OPTIONAL and REQUIRED
+# dependencies within the CMake file for that library or application.
+def getDependencyEdges(submodulePath, library):
+	cmakeListsPath = os.path.join(os.path.join(submodulePath, library), "CMakeLists.txt")
```

Comment:
Please run pep8 on the script and correct all styling errors. I would like our python scripts to adhere to the pep8 python coding style.